velox/exec/OperatorUtils.cpp

```diff
@@ -116,6 +117,26 @@ void gatherCopy(
         target, targetIndex, count, sources, sourceIndices, sourceChannel);
   }
 }
+
+// We want to aggregate some operator runtime metrics per operator rather than
+// per event. This function returns true for such metrics.
+bool shouldAggregateRuntimeMetric(const std::string& name) {
+  static const std::array<std::string, 3> metricNames{
+      "dataSourceWallNanos", "dataSourceLazyWallNanos", "queuedWallNanos"};
+  for (const auto& metricName : metricNames) {
+    if (name == metricName) {
+      return true;
+    }
+  }
+
+  // 'blocked*WallNanos'
+  if (name.size() > 16 and strncmp(name.c_str(), "blocked", 7) == 0) {
```

Comment:
use regular expression here? like static const boost::regex re("^[a-zA-Z_][a-zA-Z0-9_]*$");? Thanks!

Reply:
I believe it is not worth using re here. Using strncmp is lighter.

Reply:
But such a comparison is a bit fragile? re expression is actually easy to understand?